### PR TITLE
Add wrap files for freetype 2.9.1

### PR DESCRIPTION
**Pull request by bsmiles32**
Wrap files are based on the work done by @MathieuDuponchelle.
Minor edits regarding project license and msvc-specific warnings were added.

### CODE REVIEW

**Comment by sarum9in** [comment]
I am sorry but https://download.savannah.gnu.org/releases/freetype/freetype-2.9.1.tar.gz doesn't work.

```
$ mesonwrap review freetype2 --pull_request=1 --allow_other_files
Inspecting project freetype2, branch 2.9.1.
Repo name valid: ☑
Branch name valid: ☑
Target branch is not master: ☑
Has readme.txt: ☑
Has LICENSE.build: ☑
Has upstream.wrap: ☑
Non-buildsystem files found:
  include/freetype/config/ftoption.h.in
  builds/unix/ftconfig.h.in
Repo contains only buildsystem files: ☒
upstream.wrap has directory: ☑
upstream.wrap directory has no path separators: ☑
upstream.wrap has source_url: ☑
upstream.wrap has source_filename: ☑
upstream.wrap source_filename has no path separators: ☑
upstream.wrap has source_hash: ☑
Fetching 'https://download.savannah.gnu.org/releases/freetype/freetype-2.9.1.tar.gz'
Download url works: ☒
 error: <urlopen error [Errno -2] Name or service not known>
```

**Comment by bsmiles32** [comment]
On my machine it seems to work:
```
$ ./mesonwrap.py review freetype2 --pull_request=1 --allow_other_files
Inspecting project freetype2, branch 2.9.1.
Repo name valid: ☑
Branch name valid: ☑
Target branch is not master: ☑
Has readme.txt: ☑
Has LICENSE.build: ☑
Has upstream.wrap: ☑
Non-buildsystem files found:
  include/freetype/config/ftoption.h.in
  builds/unix/ftconfig.h.in
Repo contains only buildsystem files: ☒
upstream.wrap has directory: ☑
upstream.wrap directory has no path separators: ☑
upstream.wrap has source_url: ☑
upstream.wrap has source_filename: ☑
upstream.wrap source_filename has no path separators: ☑
upstream.wrap has source_hash: ☑
Download URL works: ☑
Hash matches: ☑
upstream.wrap directory 'freetype-2.9.1' exists: ☑
Patch merges with source: ☑
The Meson build system
Version: 0.48.2
Source dir: /tmp/tmp_y77gd0b/src/freetype-2.9.1
Build dir: /tmp/tmp_y77gd0b/bin
Build type: native build
Project name: freetype2
Project version: 22.1.16
Native C compiler: cc (gcc 8.2.1 "cc (GCC) 8.2.1 20180831")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Has header "unistd.h": YES
Has header "fcntl.h": YES
Has header "stdint.h": YES
Found pkg-config: /usr/bin/pkg-config (1.5.4)
Dependency bzip2 found: NO (tried pkgconfig)
Dependency harfbuzz found: YES 2.1.1
Dependency zlib found: YES 1.2.11
Dependency libpng found: YES 1.6.35
Configuring ftconfig.h using configuration
Configuring ftoption.h using configuration
Build targets in project: 1
Found ninja-1.8.2 at /usr/bin/ninja
meson setup: ☑
ninja: Entering directory `/tmp/tmp_y77gd0b/bin'
[28/42] Compiling C object 'freetype@sha/src_autofit_autofit.c.o'.
In file included from ../src/freetype-2.9.1/src/autofit/autofit.c:35:
../src/freetype-2.9.1/src/autofit/afshaper.c: Dans la fonction « af_shaper_get_coverage »:
../src/freetype-2.9.1/src/autofit/afshaper.c:136:5: warning: « hb_ot_tags_from_script » est obsolète: Use 'hb_ot_tags_from_script_and_language' instead [-Wdeprecated-declarations]
     hb_ot_tags_from_script( script,
     ^~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/harfbuzz/hb.h:34,
                 from ../src/freetype-2.9.1/src/autofit/afshaper.h:29,
                 from ../src/freetype-2.9.1/src/autofit/afglobal.h:26,
                 from ../src/freetype-2.9.1/src/autofit/afcjk.c:29,
                 from ../src/freetype-2.9.1/src/autofit/autofit.c:24:
/usr/include/harfbuzz/hb-deprecated.h:236:1: note: déclaré ici
 hb_ot_tags_from_script (hb_script_t  script,
 ^~~~~~~~~~~~~~~~~~~~~~
[41/42] Running all tests.
No tests defined.
ninja test: ☑
```

**Comment by sarum9in** [comment]
OK, thanks for running this. I will try to root-cause this.

**Comment by bsmiles32** [comment]
maybe proxy related ?

**Comment by sarum9in** [comment]
Wow: https://github.com/systemd/systemd/issues/9867

**Comment by nirbheek** [comment]
Maybe put an entry in `/etc/hosts` for now?

**Comment by sarum9in** [comment]
No, it's fine. I am just surprised.

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,228 @@
+project('freetype2', 'c',
+  license : ['FTL', 'GPL2'],
+  meson_version: '>= 0.47.0',
+  version: '22.1.16',
```
Comment (sarum9in) [inline meson.build]: Where this version is coming from?

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+ft_config_headers += [configure_file(input: 'include/freetype/config/ftoption.h.in',
+  output: 'ftoption.h',
+  configuration: conf)]
+
+libfreetype = library('freetype', base_sources,
+  include_directories: incbase,
+  dependencies: deps,
+  c_args: c_args,
+  install: true,
+  version: '6.16.0',
```
Comment (sarum9in) [inline meson.build]: Where this version is coming from?

File: builds/unix/ftconfig.h.in [hunk 3]
```diff
@@ -0,0 +1,603 @@
+/***************************************************************************/
```
Comment (sarum9in) [inline builds/unix/ftconfig.h.in:1]: How did you create this file?

**Review by sarum9in — CHANGES_REQUESTED** [review]
I guess you can answer the questions, but someone not so familiar with the project probably can't. Comments would help a lot for any future maintainer.

File: meson.build (hunk 1 above)
Comment (MathieuDuponchelle) [reply meson.build]: iirc that's the "libtool" version, whatever that means, we probably actually want 2.9.0 here (2.9.1? was this rebased?), see https://git.savannah.gnu.org/cgit/freetype/freetype2.git/tree/docs/VERSIONS.TXT#n56

File: meson.build (hunk 2 above)
Comment (MathieuDuponchelle) [reply meson.build]: Same as aboce, here is the CMakeLists equivalent:

https://git.savannah.gnu.org/cgit/freetype/freetype2.git/tree/CMakeLists.txt#n147

Side note: why does the README list `6.16.1`, but the latest CMakeLists still have `6.16.0` ?!

File: builds/unix/ftconfig.h.in (hunk 3 above)
Comment (MathieuDuponchelle) [reply builds/unix/ftconfig.h.in:1]: adapted copy of builds/unix/ftconfig.in , this was just too terrible to port:

https://git.savannah.gnu.org/cgit/freetype/freetype2.git/tree/CMakeLists.txt#n221

Comment (MathieuDuponchelle) [reply builds/unix/ftconfig.h.in:1]: Good question btw

File: meson.build (hunk 1 above)
Comment (sarum9in) [reply meson.build]: Please add a comment saying so. It will help a lot to anyone reading the wrap later.

File: meson.build (hunk 2 above)
Comment (sarum9in) [reply meson.build]: Please add a comment saying so. It will help a lot to anyone reading the wrap later.

File: builds/unix/ftconfig.h.in (hunk 3 above)
Comment (sarum9in) [reply builds/unix/ftconfig.h.in:1]: Please add a comment saying so. It will help a lot to anyone reading the wrap later.

File: meson.build (hunk 1 above)
Comment (bsmiles32) [reply meson.build]: Now using the release version number eg 2.9.1

File: meson.build (hunk 2 above)
Comment (bsmiles32) [reply meson.build]: Updated shared object version number eg. 6.16.1 as this is based on release 2.9.1

File: builds/unix/ftconfig.h.in (hunk 3 above)
Comment (bsmiles32) [reply builds/unix/ftconfig.h.in:1]: Added a comment indicating provenance and purpose of modifications

**Comment by bsmiles32** [comment]
$ ./mesonwrap.py review freetype2 --pull_request=1 --allow_other_files
Inspecting project freetype2, branch 2.9.1.
Repo name valid: ☑
Branch name valid: ☑
Target branch is not master: ☑
Has readme.txt: ☑
Has LICENSE.build: ☑
Has upstream.wrap: ☑
Non-buildsystem files found:
  include/freetype/config/ftoption.h.in
  builds/unix/ftconfig.h.in
Repo contains only buildsystem files: ☒
upstream.wrap has directory: ☑
upstream.wrap directory has no path separators: ☑
upstream.wrap has source_url: ☑
upstream.wrap has source_filename: ☑
upstream.wrap source_filename has no path separators: ☑
upstream.wrap has source_hash: ☑
Download URL works: ☑
Hash matches: ☑
upstream.wrap directory 'freetype-2.9.1' exists: ☑
Patch merges with source: ☑
The Meson build system
Version: 0.48.2
Source dir: /tmp/tmpqfqbiip7/src/freetype-2.9.1
Build dir: /tmp/tmpqfqbiip7/bin
Build type: native build
Project name: freetype2
Project version: 2.9.1
Native C compiler: cc (gcc 8.2.1 "cc (GCC) 8.2.1 20180831")
Build machine cpu family: x86_64
Build machine cpu: x86_64
Has header "unistd.h": YES
Has header "fcntl.h": YES
Has header "stdint.h": YES
Found pkg-config: /usr/bin/pkg-config (1.5.4)
Dependency bzip2 found: NO (tried pkgconfig)
Dependency harfbuzz found: YES 2.1.1
Dependency zlib found: YES 1.2.11
Dependency libpng found: YES 1.6.35
Configuring ftconfig.h using configuration
Configuring ftoption.h using configuration
Build targets in project: 1
Found ninja-1.8.2 at /usr/bin/ninja
meson setup: ☑
ninja: Entering directory `/tmp/tmpqfqbiip7/bin'
[29/42] Compiling C object 'freetype@sha/src_autofit_autofit.c.o'.
In file included from ../src/freetype-2.9.1/src/autofit/autofit.c:35:
../src/freetype-2.9.1/src/autofit/afshaper.c: Dans la fonction « af_shaper_get_coverage »:
../src/freetype-2.9.1/src/autofit/afshaper.c:136:5: warning: « hb_ot_tags_from_script » est obsolète: Use 'hb_ot_tags_from_script_and_language' instead [-Wdeprecated-declarations]
     hb_ot_tags_from_script( script,
     ^~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/harfbuzz/hb.h:34,
                 from ../src/freetype-2.9.1/src/autofit/afshaper.h:29,
                 from ../src/freetype-2.9.1/src/autofit/afglobal.h:26,
                 from ../src/freetype-2.9.1/src/autofit/afcjk.c:29,
                 from ../src/freetype-2.9.1/src/autofit/autofit.c:24:
/usr/include/harfbuzz/hb-deprecated.h:236:1: note: déclaré ici
 hb_ot_tags_from_script (hb_script_t  script,
 ^~~~~~~~~~~~~~~~~~~~~~
[41/42] Running all tests.
No tests defined.
ninja test: ☑

**Review by sarum9in — APPROVED** [review]

**Comment by sarum9in** [comment]
Thanks a million for adding these comments!

**Pull request merged**